mmv1/products/gkeonprem/VmwareCluster.yaml

```diff
@@ -0,0 +1,662 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: "VmwareCluster"
+min_version: beta
+base_url: "projects/{{project}}/locations/{{location}}/vmwareClusters"
+create_url: "projects/{{project}}/locations/{{location}}/vmwareClusters?vmware_cluster_id={{name}}"
+update_url: "projects/{{project}}/locations/{{location}}/vmwareClusters/{{name}}"
+delete_url: "projects/{{project}}/locations/{{location}}/vmwareClusters/{{name}}?force=true"
+self_link: "projects/{{project}}/locations/{{location}}/vmwareClusters/{{name}}"
+update_verb: :PATCH
+update_mask: true
+description: "A Google VMware User Cluster."
+autogen_async: false
+id_format: "projects/{{project}}/locations/{{location}}/vmwareClusters/{{name}}"
+import_format: ["projects/{{project}}/locations/{{location}}/vmwareClusters/{{name}}"]
+taint_resource_on_failed_create: true
+timeouts: !ruby/object:Api::Timeouts
+  insert_minutes: 60
+  delete_minutes: 60
+  update_minutes: 60
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_vmware_cluster_basic"
+    min_version: beta
+    primary_resource_id: "cluster-basic"
+    vars:
+      name: "basic"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_vmware_cluster_f5lb"
+    min_version: beta
+    primary_resource_id: "cluster-f5lb"
+    vars:
+      name: "f5lb"
+    test_env_vars:
+      project: "fake-backend-360322"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "gkeonprem_vmware_cluster_manuallb"
+    min_version: beta
+    primary_resource_id: "cluster-manuallb"
+    vars:
+      name: "manuallb"
+    test_env_vars:
+      project: "fake-backend-360322"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: "name"
+    description: The VMware cluster name.
+    immutable: true
+    url_param_only: true
+    required: true
+  - !ruby/object:Api::Type::String
+    name: "location"
+    description: The location of the resource.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: "adminClusterMembership"
+    required: true
+    description: |
+      The admin cluster this VMware User Cluster belongs to.
+      This is the full resource name of the admin cluster's hub membership.
+      In the future, references to other resource types might be allowed if
+      admin clusters are modeled as their own resources.
+    diff_suppress_func: ProjectNumberDiffSuppress
+  - !ruby/object:Api::Type::String
+    name: "description"
+    description: |
+      A human readable description of this VMware User Cluster.
+  - !ruby/object:Api::Type::String
+    name: "onPremVersion"
```

Comment:
Can this field change automatically out-of-band?

Reply:
No, this field must be updated by users.